app/code/core/Mage/Adminhtml/Block/Widget/Grid/Column/Filter/Select.php

```diff
@@ -54,7 +54,16 @@ protected function _getOptions()
      */
     protected function _renderOption($option, $value)
     {
-        $selected = (($option['value'] == $value && (!is_null($value))) ? ' selected="selected"' : '');
+        if (!array_key_exists('value', $option)) {
+            if (!array_key_exists('label', $option)) {
+                return '';
+            }
+
+            $lbl =  $this->escapeHtml($option['label']); // or just empty as well?
+            return '<option value="' . $lbl . '">' . $lbl . '</option>';
```

Comment:
What's the reason for setting label (instead of empty string) to option value?

Reply:
I did it, so each option has a different / its own value, what the 'value' key in the map was supposed to provide.
If you think that's not required, we can leave the value attribute of the tag empty.

Reply:
I'd prefer an empty string.

Maybe you can share the  extension that causes the problem, to reproduce it?

Reply:
Ok, I'll change it to empty string then.
It's an Amazon Pay add-on. I hope I can provide it in the next days

Reply:
Maybe its better to fix the extension itself. There some forks of it ...

https://github.com/kirchbergerknorr/magento1_creativestyle_amazon-payments
https://github.com/mothership-gmbh/magento1_creativestyle_amazon_payments


Or create a repo, or better update https://github.com/OpenMageModuleFostering/creativestyle_amazonpayments